src/shared/lib/StoreUtils.ts

```diff
@@ -1237,6 +1237,28 @@ export function groupBySampleId(
     }));
 }
 
+export function mapSampleIdToClinicalData<T extends { [key: string]: any }>(
+    clinicalDataGroupedBySampleId: Array<T>,
+    indexKey: keyof T,
+    clinicalDataKey: keyof T
+) {
+    const sampleIdToClinicalDataMap: {
+        [key: string]: Array<ClinicalData>;
+    } = {};
+    for (let i = 0; i < clinicalDataGroupedBySampleId.length; i++) {
+        const sampleIdKey: string = clinicalDataGroupedBySampleId[i][
+            indexKey
+        ] as string;
+        const clinicalAttributeArray: Array<
+            ClinicalData
+        > = clinicalDataGroupedBySampleId[i][clinicalDataKey] as Array<
+            ClinicalData
+        >;
+        sampleIdToClinicalDataMap[sampleIdKey] = clinicalAttributeArray;
+    }
+    return sampleIdToClinicalDataMap;
+}
+
 export function groupBy<T>(
```

Comment:
not sure where this came from originally but i believe lodash already gives us this method.

Reply:
I'm not sure either but I would hesitate to remove it because I don't know what else touches this.